common/src/stack/command/stack/commands/remove/storage/partition/__init__.py

```diff
@@ -46,49 +50,25 @@ class Command(stack.commands.remove.command,
 	"""
 
 	def run(self, params, args):
-		scope = None
+		(scope, device, mountpoint) = self.fillParams([
+			('scope', 'global'), ('device', None), ('mountpoint', None)])
 		oses = []
 		appliances = []
 		hosts = []
+		name = None
 
-		if len(args) == 0:
-			scope = 'global'
-		elif len(args) == 1:
-			try:
-				oses = self.getOSNames(args)
-			except:
-				oses = []
-
-			try:
-				appliances = self.getApplianceNames(args)
-			except:
-				appliances = []
-
-			try:
-				hosts = self.getHostnames(args)
-			except:
-				hosts = []
-		else:
-			raise ArgError(self, 'scope', 'must be unique or missing')
-
-		if not scope:
-			if args[0] in oses:
-				scope = 'os'
-			elif args[0] in appliances:
-				scope = 'appliance'
-			elif args[0] in hosts:
-				scope = 'host'
-
-		if not scope:
-			raise ParamValue(self, 'scope', 'valid os, appliance name or host name')
+		if scope == "os":
+			oses = self.getOSNames(args)
+		elif scope == "appliance":
+			appliances = self.getApplianceNames(args)
+		elif scope == "host":
+			hosts = self.getHostnames(args)
```

Comment:
This section confused me at first -- why set each of these variables and then not use them? Then I realized you were relying on the various `self.get*()`'s to raise CommandError exceptions if `args` was bad.  I'd suggest skipping creating the variables entirely.  Just call `self.get*(args)` and don't assign it to anything.

IMO, I think the original command's code was way too verbose.  Yours cuts it down, but I think you could go even further.  For example, all of the `select`'s in conditionals are using essentially the same logic.  Why not get the table name in if statement's at the beginning, and then use that when `scope != 'global'`:

```
tableid = -1
if scope != 'global':
	self.db.execute("""select id from %s where
				name = '%s' """ % (db_tablename, name))
	tableid, = self.db.fetchone()
```

This makes the whole file a lot cleaner.